### PR TITLE
Torpedo logic finished, refactoring tests and remove one singleton

### PR DESCRIPTION
**Pull request by Devas**
Added torpedo shot handling in ShotStatus class.
WindowViewTest fix - added Locale Initialization.
Added torpedo counter - dependet both from is Torpedo Shot and is One mast ship sunk.
Only torpedo logic without any connection to GUI.
Refactoring (- method naming; - code reformat; - minimize magic number;)
Removed unneccessary generator for packages - tests for sending data packages use now data providers.
Refactoring (LocaleManager removed, responsibilities moved to GameManager, using resourceBundle, polish typos fixed)

### CODE REVIEW

File: src/main/java/com/fourmusketeers/client/logic/GameManager.java [hunk 1]
```diff
@@ -23,4 +29,12 @@ public ShipsGenerator getShipsGenerator() {
...
         return shipsContainer;
     }
+
+    public void setProperLocale(Locales locale ) {
+        resource = ResourceBundle.getBundle("localization" + locale.toString());
+    }
+
+    public String get(String key) {
+        return resource.getString(key);
+    }
```
Comment (kukulastudent) [inline src/main/java/com/fourmusketeers/client/logic/GameManager.java]: enum :-1:

File: src/main/java/com/fourmusketeers/server/logic/ShotStatus.java [hunk 2]
```diff
@@ -21,17 +26,38 @@ public ShotStatus(GameTurnInfoToServer playerMove, Player opponentPlayer) {
...
+                torpedoTrajectory.add(field.getKey());
+                break;
+            } else if (field.getValue().equals(TileStatus.SHIP)) {
+                torpedoTrajectory.add(field.getKey());
+                opponentPlayer.getBoardModifier().markShipAsSunk(field.getKey());
+                opponentPlayer.removeSunkMast(field.getKey());
+                onTarget = true;
+                break;
+            }
+        }
```
Comment (kukulastudent) [inline src/main/java/com/fourmusketeers/server/logic/ShotStatus.java]: :+1:  easy for read

File: src/main/java/com/fourmusketeers/client/logic/GameManager.java (hunk 1 above)
Comment (Dzionek95) [inline src/main/java/com/fourmusketeers/client/logic/GameManager.java]: I agree

File: src/main/java/com/fourmusketeers/server/logic/Ship.java [hunk 3]
```diff
@@ -16,11 +15,11 @@ public Ship(Integer masts, List<Integer> positions) {
...
     }
 
-    public Integer getMasts() {
+    Integer getMasts() {
         return masts;
     }
 
-    public List<Integer> getPositions() {
+    List<Integer> getPositions() {
         return positions;
```
Comment (Dzionek95) [inline src/main/java/com/fourmusketeers/server/logic/Ship.java]: 👍

File: src/main/java/com/fourmusketeers/server/logic/Player.java [hunk 4]
```diff
@@ -62,5 +65,21 @@ BoardModifier getBoardModifier() {
...
+        updateTorpedoStatistic(isOneMastShip);
+    }
+
+    Map<Integer, TileStatus> torpedoTrajectory(Integer shotPosition) {
+        return board.prepareTorpedoTrajectory(shotPosition);
+    }
+
+    Integer updateTorpedoStatistic(boolean torpedoShot) {
+        return torpedoShot ? --torpedo : torpedo;
     }
```
Comment (Dzionek95) [inline src/main/java/com/fourmusketeers/server/logic/Player.java]: 👎

File: src/main/java/com/fourmusketeers/server/logic/ShotStatus.java (hunk 2 above)
Comment (Dzionek95) [inline src/main/java/com/fourmusketeers/server/logic/ShotStatus.java]: 👍

File: src/main/resources/localizationPL.properties [hunk 5]
```diff
@@ -10,4 +10,4 @@ fourMast=Czteromasztowiec:
 threeMast=Trzymasztowce:
 twoMast=Dwumasztowce:
 oneMast=Jednomasztowce:
-random=Losuj pozycje statkow
\ No newline at end of file
+random=Losowe rozmieszczenie statk\u00F3w
```
Comment (Dzionek95) [inline src/main/resources/localizationPL.properties]: 👍

File: src/test/java/com/fourmusketeers/server/ServerTest.java [hunk 6]
```diff
@@ -26,14 +27,62 @@
...
 
+    @DataProvider
+    public static Object[][] initialSetup() {
+        return new Object[][]{
+                {"Mike", generateInitialMap(2, Arrays.asList(1, 2, 78, 77))},
+                {"John", generateInitialMap(1, Arrays.asList(1, 3, 99, 64))},
+                {"Sofia", generateInitialMap(4, Arrays.asList(10, 20, 30, 40))}
+        };
+    }
+
```
Comment (Dzionek95) [inline src/test/java/com/fourmusketeers/server/ServerTest.java]: 👍

**Pull request merged**